sw/host/tests/manuf/manuf_cp_unlock_raw/src/main.rs

```diff
@@ -0,0 +1,160 @@
+    LcCtrlTransitionCtrl,
+};
+use opentitanlib::io::jtag::{Jtag, JtagParams, JtagTap};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::poll;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams,
+}
+
+// Verify transition from `RAW` to `TEST_UNLOCKED` `lc_state`.
+//
+// - [x] Pre-load OTP with `RAW` lc_state.
+// - [x] Bring device out of reset selecting the Life Cycle (LC) TAP interface using strap pins.
+//
+// Perfom the following steps via LC TAP interface:
+//
+// - [x] Switch to external clock via `TRANSITION_CTRL.EXT_CLOCK_EN`.
+// - [~] Provide `RAW_UNLOCK`.
+// - [x] Poll for lc_ctrl `STATUS` register for `TRANSITION_SUCCESSFUL`. Check for errors or timeout.
+// - [~] Perform reset by toggling the `POR` pin.
+// - [x] Verify the `lc_state` is `TEST_UNLOCKED0`.
+//
+// The manufacturing environment is required to support test reruns whenever possible:
+//
+// - [ ] Verify that the `RAW_UNLOCK` process can be re-tried after failed or interrupted attempts.
+
+fn main() -> anyhow::Result<()> {
+    let opts = Opts::from_args();
+    opts.init.init_target()?;
+
+    let transport = opts
+        .init
+        .init_target()
+        .context("failed to initialise target")?;
+
+    // Set the TAP straps for the lifecycle controller and reset.
+    transport
+        .pin_strapping("PINMUX_TAP_LC")?
+        .apply()
+        .context("failed to apply LC TAP strapping")?;
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset")?;
+
+    // Connect to the LC TAP via JTAG.
+    let jtag = transport
+        .jtag(&opts.jtag)
+        .context("failed to get JTAG interface")?;
+    jtag.connect(JtagTap::LcTap)
+        .context("failed to connect to LC TAP over JTAG")?;
+
+    lc_transition::wait_for_status(&jtag, Duration::from_secs(1), LcCtrlStatus::READY)
+        .context("failed to wait for LC to be ready")?;
+
+    // Try to claim the transitions mutex.
+    poll::poll_until(Duration::from_secs(1), Duration::from_millis(10), || {
+        jtag.write_lc_ctrl_reg(
+            &LcCtrlReg::ClaimTransitionIf,
+            u8::from(MultiBitBool8::True) as u32,
+        )
+        .context("failed to write to CLAIM_TRANSITION_IF register")?;
+
+        // Check we acquired the lock
+        let got_lock = jtag
+            .read_lc_ctrl_reg(&LcCtrlReg::ClaimTransitionIf)
+            .context("failed to read CLAIM_TRANSITION_IF register")?;
+
+        Ok(got_lock == 0x01)
+    })
+    .context("failed to claim transition mutex")?;
+
+    // Check we're currently in the `RAW` lifecycle state.
+    let state = jtag
+        .read_lc_ctrl_reg(&LcCtrlReg::LcState)
+        .context("failed to read LC_STATE register")?;
+    assert_eq!(state, u32::from(DifLcCtrlState::Raw));
+
+    // Enable the external clock.
+    jtag.write_lc_ctrl_reg(
```

Comment:
does this cause the fpga to freeze up? might comment this out, then drop a TODO here referencing #17492 and we can uncomment once external clock on FPGA is connected

Reply:
Thanks, I'll test this now

Reply:
The external clock has been enabled on FPGA now (since #17967 merged) so this hopefully does not cause the chip to be freeze up.